addons/xeh/fnc_preInit.sqf

```diff
@@ -61,15 +61,20 @@ GVAR(EventsLowercase) = [];
     };
 } forEach ([false, true] call CBA_fnc_supportMonitor);
 
-// recompile extended event handlers when enabled
-if (["xeh"] call CBA_fnc_isRecompileEnabled) then {
-    GVAR(allEventHandlers) = configFile call CBA_fnc_compileEventHandlers; // from addon config
-} else {
-    GVAR(allEventHandlers) = call (uiNamespace getVariable QGVAR(fnc_getAllEventHandlers));
-};
+// always recompile extended event handlers
+#ifdef DEBUG_MODE_FULL
+    XEH_LOG("XEH: Compiling XEH START");
+#endif
+
+GVAR(allEventHandlers) = [];
+
+{
+    GVAR(allEventHandlers) append (_x call CBA_fnc_compileEventHandlers);
+} forEach [configFile, campaignConfigFile, missionConfigFile];
```

Comment:
Originally it was `configFile -> campaignConfigFile -> missionConfigFile` too (copied this order from the old branch). Shouldn't make any difference though.